com.unity.render-pipelines.core/Runtime/Common/CoreUnsafeUtils.cs

```diff
@@ -150,6 +151,233 @@ public static void CopyTo<T>(this T[] list, void* dest, int count)
                 UnsafeUtility.WriteArrayElement<T>(dest, i, list[i]);
         }
 
+        private static void CalculateRadixParams(int radixBits, out int bitStates)
+        {
+            if (radixBits != 2 && radixBits != 4 && radixBits != 8)
+                throw new Exception("Radix bits must be 2, 4 or 8 for uint radix sort.");
+            bitStates = 1 << radixBits;
+        }
+
+        private static int CalculateRadixSupportSize(int bitStates, int arrayLength)
+        {
+            return bitStates * 3 + arrayLength;
+        }
+
+        private static unsafe void CalculateRadixSortSupportArrays(
+            int bitStates, int arrayLength, uint* supportArray,
+            out uint* bucketIndices, out uint* bucketSizes, out uint* bucketPrefix, out uint* arrayOutput)
+        {
+            bucketIndices = supportArray;
+            bucketSizes = bucketIndices + bitStates;
+            bucketPrefix = bucketSizes + bitStates;
+            arrayOutput = bucketPrefix + bitStates;
+        }
+
+        private static unsafe void MergeSort(uint* array, uint* support, int length)
+        {
+            for (int k = 1; k < length; k *= 2)
+            {
+                for (int left = 0; left + k < length; left += k * 2)
+                {
+                    int right = left + k;
+                    int rightend = right + k;
+                    if (rightend > length)
+                        rightend = length;
+                    int m = left;
+                    int i = left;
+                    int j = right;
+                    while (i < right && j < rightend)
+                    {
+                        if (array[i] <= array[j])
+                        {
+                            support[m] = array[i++];
+                        }
+                        else
+                        {
+                            support[m] = array[j++];
+                        }
+                        m++;
+                    }
+                    while (i < right)
+                    {
+                        support[m] = array[i++];
+                        m++;
+                    }
+                    while (j < rightend)
+                    {
+                        support[m] = array[j++];
+                        m++;
+                    }
+                    for (m = left; m < rightend; m++)
+                    {
+                        array[m] = support[m];
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Merge sort - non recursive
+        /// </summary>
+        /// <param name="arr">array to sort.</param>
```

Comment:
Super minor nitpick, but descriptions should all start capitalized (e.g. **A**rray to sort). 

Same applies to lot of things in this file, gonna only mark this here :)

Reply:
Done!